src/js/pages/services/ServicesTab.js

```diff
@@ -0,0 +1,223 @@
+import React from 'react';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import AlertPanel from '../../components/AlertPanel';
+import Config from '../../config/Config';
+import DCOSStore from '../../stores/DCOSStore';
+import FilterHeadline from '../../components/FilterHeadline';
+import SaveStateMixin from '../../mixins/SaveStateMixin';
+import Service from '../../structs/Service';
+import ServiceDetail from '../../components/ServiceDetail';
+import ServiceFilterTypes from '../../constants/ServiceFilterTypes';
+import ServiceSidebarFilters from '../../components/ServiceSidebarFilters';
+import ServicesTable from '../../components/ServicesTable';
+import ServiceTree from '../../structs/ServiceTree';
+import SidebarActions from '../../events/SidebarActions';
+import SidePanels from '../../components/SidePanels';
+
+var DEFAULT_FILTER_OPTIONS = {
+  filterHealth: null,
+  searchString: ''
+};
+
+let saveState_properties = Object.keys(DEFAULT_FILTER_OPTIONS);
+
+var ServicesTab = React.createClass({
+
+  displayName: 'ServicesTab',
+
+  saveState_key: 'servicesPage',
+
+  saveState_properties,
+
+  mixins: [SaveStateMixin, StoreMixin],
+
+  statics: {
+    // Static life cycle method from react router, that will be called
+    // 'when a handler is about to render', i.e. on route change:
+    // https://github.com/rackt/react-router/
+    // blob/master/docs/api/components/RouteHandler.md
+    willTransitionTo: function () {
+      SidebarActions.close();
+    }
+  },
+
+  contextTypes: {
+    router: React.PropTypes.func
+  },
+
+  getInitialState: function () {
+    return Object.assign({}, DEFAULT_FILTER_OPTIONS);
+  },
+
+  componentWillMount: function () {
+    this.store_listeners = [{name: 'dcos', events: ['change']}];
+  },
+
+  handleFilterChange: function (filterValues, filterType) {
+    var stateChanges = Object.assign({}, DEFAULT_FILTER_OPTIONS);
+    stateChanges[filterType] = filterValues;
+
+    this.setState(stateChanges);
+  },
+
+  resetFilterQueryParams: function () {
+    let router = this.context.router;
+    let queryParams = router.getCurrentQuery();
+
+    Object.values(ServiceFilterTypes).forEach(function (filterKey) {
```

Comment:
I understand you may not have written this code, but we should still change it. We shouldn't be merging anything that has show stoppers in it, irrespective of who wrote it. :)